tools/find_deprecated.py

```diff
@@ -236,6 +249,34 @@ def find_deprecations(file_name: Path) -> list[Deprecation]:
                     DETAILS += f" (wrt last minor release, {round(diff_days / 30.4)} month old)"
             except KeyError:
                 DETAILS = "Future release"
-        print(f"\n{since_version}: {DETAILS}")
+        lines = []
         for deprecation in deprecations:
-            print(f" - {deprecation.location_str} ({deprecation.target})")
+            is_pending = " - PENDING" if deprecation.pending else ""
+            if pending == "include":
+                lines.append(f" - {deprecation.location_str} ({deprecation.target}){is_pending}")
+            if pending == "only" and deprecation.pending:
+                lines.append(f" - {deprecation.location_str} ({deprecation.target}){is_pending}")
+            if pending == "exclude" and not deprecation.pending:
+                lines.append(f" - {deprecation.location_str} ({deprecation.target})")
+        if lines:
+            print(f"\n{since_version}: {DETAILS}")
+            print("\n".join(lines))
+
+
+if __name__ == "__main__":
+    default_directory = Path(__file__).joinpath("..", "..", "qiskit").resolve()
+    parser = argparse.ArgumentParser(
```

Comment:
Nit: I really like the pattern of having a `create_parser() -> ArgumentParser` helper function. Then your `main` looks like `create_parser().parse_args()`. Makes it more readable imo.